### PR TITLE
lending: Add SetLendingMarketOwner instruction

### PR DESCRIPTION
**Pull request by jstarry**
#### Problem
No way to migrate an existing lending market to a new owner. In this case, I would like to move the owner authority to a multisig contract

### CODE REVIEW

File: token-lending/program/src/instruction.rs [hunk 1]
```diff
@@ -78,44 +87,53 @@ pub enum LendingInstruction {
...
+    ///
     ///   0. `[writable]` Source liquidity token account. $authority can transfer $liquidity_amount
     ///   1. `[writable]` Destination collateral token account.
     ///   2. `[writable]` Reserve account.
     ///   3. `[writable]` Reserve liquidity supply SPL Token account.
     ///   4. `[writable]` Reserve collateral SPL Token mint.
     ///   5. `[]` Lending market account.
     ///   6. `[]` Derived lending market authority.
-    ///   7. `[]` User transfer authority ($authority).
+    ///   7. `[signer]` User transfer authority ($authority).
```
Comment (jordansexton) [inline token-lending/program/src/instruction.rs:103]: Nice

File: token-lending/program/src/processor.rs [hunk 2]
```diff
@@ -1497,6 +1501,33 @@ fn process_withdraw_obligation_collateral(
...
+        return Err(LendingError::InvalidAccountOwner.into());
+    }
+    if &lending_market.owner != lending_market_owner_info.key {
+        return Err(LendingError::InvalidMarketOwner.into());
+    }
+    if !lending_market_owner_info.is_signer {
+        return Err(LendingError::InvalidSigner.into());
+    }
+
+    lending_market.owner = new_owner;
```
Comment (jordansexton) [inline token-lending/program/src/processor.rs:1525]: Should this error if the lending market owner is already `new_owner`? (I don't know how no-ops are generally handled)

File: token-lending/program/tests/set_lending_market_owner.rs [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        Some(&payer.pubkey()),
+    );
+
+    transaction.sign(&[&payer, &lending_market.owner], recent_blockhash);
+
+    banks_client
+        .process_transaction(transaction)
+        .await
+        .map_err(|e| e.unwrap())
+        .unwrap();
```
Comment (jordansexton) [inline token-lending/program/tests/set_lending_market_owner.rs:50]: What does this do if no error is expected / what's the difference between this and

```rust
assert!(banks_client.process_transaction(transaction).await.is_ok());
```

**Review by jordansexton — APPROVED** [review]
Looks good, I like the additional instruction docs. Had two questions but non blocking.

File: token-lending/program/src/processor.rs (hunk 2 above)
Comment (jstarry) [reply token-lending/program/src/processor.rs:1525]: I don't think there's any downside to allowing a no-op. The SPL token SetAuthority instruction also allows no-ops

File: token-lending/program/tests/set_lending_market_owner.rs (hunk 3 above)
Comment (jstarry) [reply token-lending/program/tests/set_lending_market_owner.rs:50]: If it errors, it will panic due to the unwrap. An assert is better, thanks!

Comment (jstarry) [reply token-lending/program/tests/set_lending_market_owner.rs:50]: Oops, my patch for the assert didn't get pushed before the auto-merge label took effect. I'll clean this up later, promise 😉

**Pull request merged**